l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1220,6 +1226,20 @@ def importFatturaPA(self):
                 new_invoices.append(invoice_id)
                 self.check_invoice_amount(invoice, fattura)
 
+                amount_untaxed = self.compute_xml_amount_untaxed(
+                    fattura.DatiBeniServizi.DatiRiepilogo)
+                amount_tax = self.compute_xml_amount_tax(
+                    fattura.DatiBeniServizi.DatiRiepilogo)
+                amount_total = float(
+                    fattura.DatiGenerali.DatiGeneraliDocumento.
+                    ImportoTotaleDocumento or 0.0)
+
+                invoice.update({
+                    'e_invoice_amount_untaxed': amount_untaxed,
+                    'e_invoice_amount_tax': amount_tax,
+                    'e_invoice_amount_total': amount_total,
+                })
```

Comment:
@robyf70 ieri analizzando i vari casi d'uso, ci siamo accorti che questi controlli sarebbero utili anche quando la fattura viene generata da odoo (ad esempio da un PO) e successivamente collegata ad un XML.

Questi campi potrebbero quindi essere scritti anche in fase di "collegamento", nel wizard `wizard.link.to.invoice`.
Cosa ne pensi?

Reply:
@eLBati Penso che questo completerebbe i casi supportati adesso. Potremmo inserire qualcosa tipo:
```diff
@@ -16,6 +16,33 @@ class WizardLinkToInvoice(models.TransientModel):
         xml_string = fatturapa_attachment.get_xml_string()
         return fatturapa_v_1_2.CreateFromDocument(xml_string)
 
+    def compute_xml_amount_untaxed(self, DatiRiepilogo):
+        amount_untaxed = 0.0
+        for Riepilogo in DatiRiepilogo:
+            amount_untaxed += float(Riepilogo.ImponibileImporto)
+        return amount_untaxed
+
+    def compute_xml_amount_tax(self, DatiRiepilogo):
+        amount_tax = 0.0
+        for Riepilogo in DatiRiepilogo:
+            amount_tax += float(Riepilogo.Imposta)
+        return amount_tax
+
+    def set_einvoice_amount(self, fattura):
+        amount_untaxed = self.compute_xml_amount_untaxed(
+            fattura.DatiBeniServizi.DatiRiepilogo)
+        amount_tax = self.compute_xml_amount_tax(
+            fattura.DatiBeniServizi.DatiRiepilogo)
+        amount_total = float(
+            fattura.DatiGenerali.DatiGeneraliDocumento.
+            ImportoTotaleDocumento or 0.0)
+
+        self.invoice_id.update({
+            'e_invoice_amount_untaxed': amount_untaxed,
+            'e_invoice_amount_tax': amount_tax,
+            'e_invoice_amount_total': amount_total,
+        })
+
     @api.multi
     def link(self):
         self.ensure_one()
@@ -29,6 +56,7 @@ class WizardLinkToInvoice(models.TransientModel):
             fatturapa_attachment = fatturapa_attachment_obj.browse(
                 fatturapa_attachment_id)
             fatt = self.get_invoice_obj(fatturapa_attachment)
+            self.set_einvoice_amount(fatt)
             for FatturaBody in fatt.FatturaElettronicaBody:
                 # 2.5
                 AttachmentsData = FatturaBody.Allegati
```

Reply:
:+1:

Reply:
@eLBati La patch era leggermente diversa, ho appena committato quella funzionante